### PR TITLE
[quantization] Refactoring names for consistency

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27779 [quantization] Changing observer name
* **#27670 [quantization] Refactoring names for consistency**

Differential Revision: [D17846269](https://our.internmc.facebook.com/intern/diff/D17846269)

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]
this reminds me that we need to rename 'self.observer' to 'self.activation_post_process' do we want to do it now?

**Review by jianyuh — APPROVED** [review]
LGTM!

**Comment by zafartahirov** [comment]
> this reminds me that we need to rename 'self.observer' to 'self.activation_post_process' do we want to do it now?

Might as well

**Comment by zafartahirov** [comment]
> this reminds me that we need to rename 'self.observer' to 'self.activation_post_process' do we want to do it now?

Wait, in which context?

**Comment by zafartahirov** [comment]
> this reminds me that we need to rename 'self.observer' to 'self.activation_post_process' do we want to do it now?

Created a new PR: #27779

File: docs/source/quantization.rst [hunk 1]
```diff
@@ -190,10 +190,10 @@ Layers for the quantization-aware training
     * :class:`~torch.quantization.Observer` — Abstract base class for observers
 * Quantization configurations
     * :class:`~torch.quantization.QConfig` — Quantization configuration class
-    * ``default_qconfig`` — Same as ``QConfig(activation=default_observer, weight=default_weight_observer)`` (See :class:`~torch.quantization.QConfig.QConfig`)
-    * :attr:`~torch.quantization.default_qat_qconfig` — Same as ``QConfig(activation=default_fake_quant, weight=default_weight_fake_quant)`` (See :class:`~torch.quantization.QConfig.QConfig`)
-    * :attr:`~torch.quantization.default_dynamic_qconfig` — Same as ``QConfigDynamic(weight=default_weight_observer)`` (See :class:`~torch.quantization.QConfig.QConfigDynamic`)
-    * :attr:`~torch.quantization.float16_dynamic_qconfig` — Same as ``QConfigDynamic(weight=NoopObserver.with_args(dtype=torch.float16))`` (See :class:`~torch.quantization.QConfig.QConfigDynamic`)
+    * :attr:`~torch.quantization.default_qconfig` — Same as ``QConfig(activation=default_observer, weight=default_weight_observer)`` (See :class:`~torch.quantization.qconfig.QConfig`)
```
Comment (zou3519) [inline docs/source/quantization.rst]: `default_qconfig` doesn't exist as an attribute, so it is better to label it as ``default_qconfig``
image

Comment (z-a-f) [reply docs/source/quantization.rst]: I will send a PR to set it as just verbatim text.

**Pull request closed without merging**